src/nvim/buffer_updates.c

```diff
@@ -281,9 +288,17 @@ void buf_updates_send_changes(buf_T *buf, linenr_T firstline, int64_t num_added,
         args.items[6] = INTEGER_OBJ((Integer)deleted_codepoints);
         args.items[7] = INTEGER_OBJ((Integer)deleted_codeunits);
       }
+
+      pos_T save_cursor;
+      if (curwin != NULL) {
```

Comment:
is `curwin` being null in a text change something that ever happens?

Reply:
When I created this PR unload callbacks were called on exit. It guess it shouldn't happen now.